pkg/database/key_server_stats.go

```diff
@@ -92,6 +92,19 @@ func (kssd *KeyServerStatsDay) BeforeSave(tx *gorm.DB) error {
 	return kssd.ErrorOrNil()
 }
 
+func (kssd *KeyServerStatsDay) TotalPublishRequests() int64 {
+	var sum int64
+	for _, v := range kssd.PublishRequests {
+		sum += v
+	}
+	return sum
+}
+
+func (db *Database) HasKeyServerStats(realmID uint) bool {
```

Comment:
I'd suggest we drop this and have callers check the result of GetKeyServerStats. There's also an edge case here because this checks the live data, but then the code later checks the cached data.

Reply:
GetKeyServerStats is checking for the existence of the row in the KeyServerStats table

The listing of the stats is reading KeyServerStatsDay

When we disable key server stats, we don't garbage collect old data - so presence of old data doesn't indicate if we should show it or not.